incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -268,7 +303,36 @@ private void onMqttBegin(
 
             assert initialAck <= initialSeq;
 
-            delegate.doKafkaBegin(traceId, authorization, affinity);
+            final OctetsFW extension = begin.extension();
+            final MqttBeginExFW mqttBeginEx = extension.get(mqttBeginExRO::tryWrap);
+
+            assert mqttBeginEx.kind() == MqttBeginExFW.KIND_SUBSCRIBE;
+            final MqttSubscribeBeginExFW mqttSubscribeBeginEx = mqttBeginEx.subscribe();
+
+            clientId = mqttSubscribeBeginEx.clientId().asString();
+            currentFilters = mqttSubscribeBeginEx.filters();
```

Comment:
Why do we need to keep these filters around as fields? Can they live on stack instead?

Note that the reference refers to flyweights backed by message from streams ring buffer, so becomes invalid when stack unwinds.

Reply:
This is somewhat similar to the deferred retained filters. If we can come up with a different solution for that, this might be solved differently as well.

Reply:
Of course, I have the Subscription object for that

Reply:
We can't use the same solution as we did for the deferred retained filters, as we decided to solve the issue using Kafka flush. Since we first need to server the retained messages, we need to store the filters. I'm using flyweights for that, but using a buffer per client.